library/core/src/iter/traits/iterator.rs

```diff
@@ -719,24 +717,17 @@ pub trait Iterator {
 
     /// Creates an iterator that both filters and maps.
     ///
-    /// The closure must return an [`Option<T>`]. `filter_map` creates an
-    /// iterator which calls this closure on each element. If the closure
-    /// returns [`Some(element)`][`Some`], then that element is returned. If the
-    /// closure returns [`None`], it will try again, and call the closure on the
-    /// next element, seeing if it will return [`Some`].
+    /// Given an element the closure must return `Some(value)` or `None`. The
+    /// returned iterator will only yield a value when the closure returns
+    /// `Some`. The values inside the `Some`s are yielded.
     ///
-    /// Why `filter_map` and not just [`filter`] and [`map`]? The key is in this
-    /// part:
+    /// `filter_map` can be used to make chains of [`filter`] and [`map`] more
+    /// concise. The example below shows how a `map().filter().map()` can be
+    /// shortened to a single call to `filter_map`.
```

Comment:
This is a lot better, but still leaves most of the discussion to the example instead of the prose. Maybe that's not a bad thing?

Reply:
That was my thinking. The existing example is really good, and extra words just dilutes it.

Reply:
Sounds good to me!